### PR TITLE
Updated Beachfront bidder

### PR DESCRIPTION
**Pull request by RossGoncharuk**
Refactored Beachfront bidder to include all recent changes.

@rpanchyk & @DGarbar please note that I've decided to apply some changes differently, in the way that doesn't change bidder's flow, but reduces the amount of operations, i.e. changes to some models, their usage, logical operations, etc.

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java [hunk 1]
```diff
@@ -499,42 +443,66 @@ private static Bid updateVideoBid(Bid bid, BeachfrontVideoImp imp, String id) {
...
+
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+
+        final List<Bid> bids = bidResponse.getSeatbid().get(0).getBid();
+        final List<Imp> imps = videoRequest.getRequest().getImp();
+        if (httpRequest.getUri().contains(NURL_VIDEO_ENDPOINT_SUFFIX)) {
+            return Result.of(updateVideoBids(bids, imps).stream()
+                            .map(bid -> BidderBid.of(bid, BidType.video, DEFAULT_BID_CURRENCY))
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java]: They also use `TypedBid.BidVideo` which we couldn't have.

File: src/main/java/org/prebid/server/bidder/beachfront/model/BeachfrontVideoRequest.java [hunk 2]
```diff
@@ -18,15 +15,7 @@
...
-    String id;
-
-    List<BeachfrontVideoImp> imp;
-
-    Site site;
-
-    BeachfrontVideoDevice device;
-
-    User user;
+    String videoResponseType;
```
[the remark this answers is not in the document]
Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/bidder/beachfront/model/BeachfrontVideoRequest.java]: Thanks!

File: src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java [hunk 3]
```diff
@@ -72,170 +75,186 @@ public BeachfrontBidder(String bannerEndpointUrl, String videoEndpointUrl) {
...
         }
 
-        if (latestExtImpBeachfront == null) {
-            errors.add("No valid impressions were found");
-            throw new InvalidRequestException(errors);
+        final String videoAppId = appIds != null ? appIds.getVideo() : null;
+        if (StringUtils.isNotBlank(videoAppId)) {
+            return appIds.getVideo();
         }
+        throw new PreBidException("unable to determine the appId(s) from the supplied extension");
```
[the remark this answers is not in the document]
Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java]: I guessed that too, I think the main point was to check if there's an error present then to `continue` i.e. to skip the current imp due to invalid appId.

File: src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java [hunk 4]
```diff
@@ -249,237 +268,162 @@ private static User makeUserForVideoRequest(User user) {
...
-                    beachfrontSlots.add(BeachfrontSlot.of(null, null, imp.getBidfloor(), sizes));
-                    continue;
+                secure = getSecure(site.getPage());
+            }
+
+            final App app = bidRequest.getApp();
+            if (app != null && StringUtils.isBlank(app.getDomain()) && StringUtils.isNotBlank(app.getBundle())) {
+                String trimmedBundle = app.getBundle();
+
+                if (trimmedBundle.startsWith("_")) {
```
[the remark this answers is not in the document]
Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java]: Yep, that's why I didn't use `String.trim()`. I did basically the same what `StringUtils.removeStart/End` does, just in explicit way. Will reuse the utility methods instead.

File: src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java [hunk 5]
```diff
@@ -249,237 +268,162 @@ private static User makeUserForVideoRequest(User user) {
...
-            beachfrontRequestsBuilder.domain(app.getDomain());
-            beachfrontRequestsBuilder.page(app.getId());
-        } else {
-            if (site != null) {
-                beachfrontRequestsBuilder.domain(HttpUtil.getDomainFromUrl(site.getPage()));
-                beachfrontRequestsBuilder.page(site.getPage());
+            final Device device = bidRequest.getDevice();
+            if (device != null) {
+                final Device.DeviceBuilder deviceBuilder = device.toBuilder();
+                if (device.getDevicetype() == null) {
```
[the remark this answers is not in the document]
Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java]: Done

File: src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java (hunk 1 above)
Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java]: I know, I noticed that object, but we simply don't have one so I've left it as it is. Dunno how crucial is this, need @rpanchyk to take a look.

Comment (rpanchyk) [reply src/main/java/org/prebid/server/bidder/beachfront/BeachfrontBidder.java]: that's ok, `BidType.video` is fine.

**Pull request merged**